src/nrnmpi/bbsmpipack.cpp

```diff
@@ -109,7 +109,7 @@ void nrnmpi_upkbegin(bbsmpibuf* r) {
            r->keypos);
 #endif
     assert(r && r->buf && r->size > 0);
-    if (nrnmpi_myid_bbs == -1) {
+    if ((nrnmpi_myid_bbs >= 0) && (nrnmpi_myid != 0)) {
```

Comment:
If I understood correctly with your changes you suggest `nrnmpi_myid_bbs` being the same as the rank in the `MPI_COMM_WORLD` for which `nrnmpi_myid == 0`. This means that with my suggestion below to change lines 270 and 271 of `src/nrnmpi/nrnmpi.cpp` `nrnmpi_myid_bbs` will always be `>0` and we should only check for `nrnmpi_myid != 0`?

Reply:
I was not completely comfortable with eliminating the condition on nrnmpi_myid_bbs, but it does look like it would be always greater than or equal to 0. Let me revisit this when the subworld tests are passing.

Reply:
No longer relevant, as I have reverted the changes to _bbs variable semantics.